src/query/remote/compressed_codecs.go

```diff
@@ -164,7 +164,7 @@ func CompressedSeriesFromSeriesIterator(
 	compressedReplicas := make([]*rpc.M3CompressedValuesReplica, 0, len(replicas))
 	for _, replica := range replicas {
 		replicaSegments := make([]*rpc.M3Segments, 0, len(replicas))
-		readers := replica.Readers()
+		readers := replica.Readers().Clone()
```

Comment:
This will need to get released now if it's creating an iterator

Reply:
Hm what do you mean by release? As far as I can tell it really is just a wrapping struct. And we don't want to close the internal block readers because those are shared between original and clone

Reply:
e.g. `readerSliceOfSlicesIterator.Clone()` creates a new ReaderSliceOfSlicesIterator with a backing pool; we need to be able to release that back to the pool, but that may be difficult if it has a lifecycle tied to the underlying data

Reply:
Ah I see now so on `newReaderSliceOfSlicesIterator`

Reply:
Well so since we are creating our own wrapper not from the pool, we shouldn't need to put it back.

Reply:
I'm suggesting we don't use clone to avoid answering these questions :p

Reply:
Yeah that is fair. But there are complications on the other side as well. E.g. there is a `singleSlicesOfSlicesIterator` implementation which doesn't actually have an index - it is just a bool of `firstNext`.